kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/InCommitTimestampSuite.scala

```diff
@@ -0,0 +1,150 @@
+    for (i <- 0 until commitInfoVector.getSize) {
+      if (!commitInfoVector.isNullAt(i)) {
+        val commitInfo = CommitInfo.fromColumnVector(commitInfoVector, i)
+        if (commitInfo != null && commitInfo.getInCommitTimestamp.isPresent) {
+          return Optional.of(commitInfo.getInCommitTimestamp.get)
+        }
+      }
+    }
+    Optional.empty()
+  }
+
+  test("Enable ICT on commit 0") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val beforeCommitAttemptStartTime = System.currentTimeMillis
+      val table = Table.forPath(engine, tablePath)
+
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = true, IN_COMMIT_TIMESTAMPS_ENABLED, "true", true)
+
+      val ver0Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assert(getInCommitTimestamp(engine, table, 0).get >= beforeCommitAttemptStartTime)
+      assert(ver0Snapshot.getProtocol.getWriterFeatures.contains("inCommitTimestamp-preview"))
+    }
+  }
+
+  test("Create a non-inCommitTimestamp table and then enable timestamp") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
+
+      val txn1 = txnBuilder
+        .withSchema(engine, testSchema)
+        .build(engine)
+
+      txn1.commit(engine, emptyIterable())
+
+      val ver0Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver0Snapshot, TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED, false)
+      assert(!ver0Snapshot.getProtocol.getWriterFeatures.contains("inCommitTimestamp-preview"))
+      assert(!getInCommitTimestamp(engine, table, 0).isPresent)
+
+      val beforeCommitAttemptStartTime = System.currentTimeMillis
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = false, IN_COMMIT_TIMESTAMPS_ENABLED, "true", true)
+
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assert(ver1Snapshot.getProtocol.getWriterFeatures.contains("inCommitTimestamp-preview"))
+      assert(
+        getInCommitTimestamp(engine, table, 1).get >= beforeCommitAttemptStartTime)
+    }
+  }
+
+  test("Enablement tracking properties should not be added if ICT is enabled on commit 0") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = TableImpl.forPath(engine, tablePath)
+
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = true, IN_COMMIT_TIMESTAMPS_ENABLED, "true", true)
+
+      val ver0Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver0Snapshot, TableConfig.IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP,
+        Optional.empty())
+      assertMetadataProp(ver0Snapshot, TableConfig.IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION,
+        Optional.empty())
+    }
+  }
+
+  test("Enablement tracking works when ICT is enabled post commit 0") {
```

Comment:
Can you expand this test case so that it checks that the enablement properties don't change if you add more commits?